packages/map-template/src/components/LocationsList/LocationsList.jsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import './LocationsList.scss';
+import { ReactComponent as CloseIcon } from '../../assets/close.svg';
+import ListItemLocation from "../WebComponentWrappers/ListItemLocation/ListItemLocation";
+
+/**
+ * Show list of locations.
+ *
+ * @param {Object} props
+ * @param {function} props.onLocationClick - Function that is run when a location is clicked.
+ * @param {function} props.onBack - Function that is run when the user navigates to the previous page.
+ * @param {array} props.filteredLocationsByExternalIDs - Array of locations filtered based on the external ID.
+ * @param {function} props.onLocationsFilteredByExternalIDs - Function that handles the filtered locations by external ID on the map.
+ *
+ */
+function LocationsList({ onBack, onLocationClick, filteredLocationsByExternalIDs, onLocationsFilteredByExternalIDs }) {
```

Comment:
To make this component even more generic, we should avoid mentioning external ID's.

- `filteredLocationsByExternalId` -> `locations`
- Move the assignment of filtered locations to [] outside of the component, maybe in the function running based on `onBack`.

Reply:
Refactored the Locations List component and moved the resetting of the filters inside the `Bottom sheet` and the `Sidebar` components.